### ISSUE TITLE
model train

### ISSUE THREAD

**Issue by AidtuoliAjou** [google-deepmind/md4#1]
hellow，When using the OpenWebText dataset for training, is the number of training steps one million? The SEED model was only trained for four hundred thousand steps. Once training on the OpenWebText dataset is finished, how should the perplexity be computed, or is it possible to provide the loss value for the OpenWebText dataset?

**Comment by AidtuoliAjou** [comment]
It say only need four hundred thousand steps
[https://github.com/louaaron/Score-Entropy-Discrete-Diffusion/issues/6](url)

**Comment by AidtuoliAjou** [comment]
It said only need four hundred thousand steps
![屏幕截图_5-1-2025_20250_github com](https://github.com/user-attachments/assets/4b0c396e-6b16-48f9-9f3d-af85ee57bc06)

**Comment by thjashin** [comment]
Hi @AidtuoliAjou, it was unclear to us when we were conducting the experiment how many iterations SEDD used so we reimplemented it in JAX and used the same number of iterations as text8 (1m) - please refer to the SEDD Absorb (reimpl.) in the paper.

**Comment by thjashin** [comment]
For validation perplexity computed on OpenWebText please refer to Table 5 in our Appendix (this is calculated in our training code).

**Comment by AidtuoliAjou** [comment]
OK，thanks

**Comment by kkj15dk** [comment]
Is the code for SEDD Absorb (reimpl.) present in this repository?
Maybe I am just not able to find it, if so sorry for that!
I'm mainly interested because the pytorch implementation of SEDD does not seem to have the same training instabilities as seen in figure 5 of your paper.
